### PR TITLE
update constitution

### PR DESCRIPTION
**Pull request by bytemaster**

### CODE REVIEW

File: governance/constitution.md [hunk 1]
```diff
@@ -1,44 +1,58 @@
+This Constitution is a multi-party contract entered into by the Members by virtue of their use of this blockchain. 
 
-This constitution is a multi-party contract entered into by the Members by virture of their use of this blockchain. 
-
-# Article I
+# Article I - Non Violence
```
Comment (jgiszczak) [inline governance/constitution.md]: Nonviolence

File: governance/constitution.md [hunk 2]
```diff
@@ -1,44 +1,58 @@
...
+# Article II - Perjury
+Members shall be liable for losses caused by false or misleading attestations and shall forfeit any profit gained thereby. 
 
-# Article IV
-The Members agree to resolve disputes through the blockchain’s arbitration process, or any other process that the parties to a transaction may mutually agree.
+# Article III - Rights
+The Members grant the right of contract and of private property to each other, therefore no property shall change hands except with the consent of the owner, by a valid Arbitrator’s order, or via community referendum. This Constitution creates no positive rights for or between any Members.
 
-# Article V
+# Article IV - Vote Buying
```
Comment (jgiszczak) [inline governance/constitution.md]: No Vote Buying

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 3]
```diff
@@ -1,15 +1,63 @@
...
+The community agrees to allow {{producer}} to authenticate peers as necessary to prevent abuse and denial of service attacks; however, {{producer}} agrees not to discriminate against non-abusive peers.
+
+I agree to process transactions on a FIFO best-effort and to honestly bill transactions for measured execution time. 
+
+I {{producer}} agree not to manipulate the contents of blocks in order to derive profit from:
+
+1. the order in whcih transactions are included
+2. the hash of the block that is produced 
+
+I, {{producer}}, hereby agree to disclose and attest under penalty of purgery all ultimate beneficial owners of my company who own more than 1%.
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: sp: perjury

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 4]
```diff
@@ -1,15 +1,63 @@
...
 
-## regproducer
-    ( const account_name producer
-    , const public_key& producer_key
-    , const std::string& url
-    , uint16_t location );
+If {{producer}} is selected to produce blocks by the eosio contract, I will sign blocks with {{producer_key}} and I hereby certify that I will keep this key secret and secure. 
 
-_Intent: create a producer-config and producer-info object for 'producer'_
+If {{producer}} is unable to perform obligations under this contract I will resign my position by setting by resubmitting this contract with
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: looks like half of this string should be removed: "by setting by resubmitting"

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 5]
```diff
@@ -1,15 +1,63 @@
...
+{{producer}} hereby agrees to only use {{producer_key}} to sign messages under the following scenarios:
 
-All disputes arising from this contract shall be resolved in the EOS Core Arbitration Forum. 
+1. proposing a block at the time appointed by the block scheduling algorithm 
+2. pre-confirmation for  block produced by a 3rd party when it is deemed valid 
+3. confirming a block for which {{producer}} has received 2/3+ pre-confirmation messages from other producers
+
+I hereby accept liability for any and all provable damages that result from:
+
+1. signing two different block proposals with the same timestamp with {{producer_key}
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: Missing the ending curly brace for producer key "}"

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 6]
```diff
@@ -1,15 +1,63 @@
...
+2. signing two different block proposals with the same block number with {{producer_key}
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: Missing the ending curly brace for producer key "}"

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 7]
```diff
@@ -1,15 +1,63 @@
...
+3. signing any block proposal which builds off of an objectively bad block
+4. signing a pre-confirmation for an objectively invalid block according to deterministic blockchain rules
+5. signing a confirmation for a block for which I do not possess pre-confirmation messages from 2/3+ other producers
+
+I hereby agree that double-signing for a timestamp or block number in concert with 2 or more other producers shall automatically be deemed malicious and subject to a fine equal to the past year of compensation received and imediate disqualification from being a producer. An exception may be made if {{producer}} can demonstrate that the double-signing occured due to a bug in the reference software; the burden of proof is on {{producer}}.
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: sp: occurred and immediate are misspelled

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 8]
```diff
@@ -1,15 +1,63 @@
...
+
+I hereby agree not to interfer with the election process and to sign any blocks or confirmations necessary to facilitate transfer of control to the next set of producers as determined by the system contract.
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: sp: interfere

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 9]
```diff
@@ -1,15 +1,63 @@
...
+
+If {{producer}} qualifies to receive compensation due to votes received, {{producer}} will provide a public endpoint allowing at least 100 peers to maintain synchronization with the blockchain and/or submit transactions to be included. {{producer}} shall maintain at least 1 validating node validating with full state and signature checking and shall report any objectively invalid blocks produced by the active block producers.
+
+The community agrees to allow {{producer}} to authenticate peers as necessary to prevent abuse and denial of service attacks; however, {{producer}} agrees not to discriminate against non-abusive peers.
+
+I agree to process transactions on a FIFO best-effort and to honestly bill transactions for measured execution time. 
+
+I {{producer}} agree not to manipulate the contents of blocks in order to derive profit from:
+
+1. the order in whcih transactions are included
```
Comment (robrigo) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: sp: which

File: contracts/eosio.system/eosio.system-regproducer-rc.md [hunk 10]
```diff
@@ -1,15 +1,63 @@
...
+
+I, {{producer}}, agree to set {{location}} such that {{producer}} is scheduled with minimal latency between my previous and next peer.
+
+I, {{producer}}, agree to maintain time synchronization within 10 ms of global atomic clock time.
+
+I, {{producer}}, agree not to produce blocks before my scheduled time unless I have received all blocks produced by the prior producer.
+
+I, {{producer}}, agree not to publish blocks with timestamps more than 500ms in the future unless the prior block is more than 75% full by either CPU or network bandwidth metrics. 
+
+I, {{producer}}, agree not to set the RAM supply to more RAM than my nodes contain and to resign if I am unable to provide the RAM approved by 2/3+ producers.
```
Comment (eny-rs) [inline contracts/eosio.system/eosio.system-regproducer-rc.md]: Is it meant to be 2/3+1?

File: contracts/eosio.system/eosio.system-regproducer-rc.md (hunk 3 above)
Comment (eny-rs) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: Remove the word "beneficial" to not confuse with US legal/financial terms

File: contracts/eosio.system/eosio.system-regproducer-rc.md (hunk 4 above)
Comment (thomasbcox) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: yes

File: contracts/eosio.system/eosio.system-regproducer-rc.md (hunk 5 above)
Comment (thomasbcox) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: good

File: contracts/eosio.system/eosio.system-regproducer-rc.md (hunk 6 above)
Comment (thomasbcox) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: Got it

File: contracts/eosio.system/eosio.system-regproducer-rc.md (hunk 10 above)
Comment (thomasbcox) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: We routinely use the string "2/3+" to signal "two thirds plus one"

Comment (eny-rs) [reply contracts/eosio.system/eosio.system-regproducer-rc.md]: OK

**Pull request merged**